[CI] IndexStatsIT.testConcurrentIndexingAndStatsRequests failure in 7.6

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.6+multijob-unix-compatibility/os=debian-9&&immutable/65/console

```
01:43:58 org.elasticsearch.indices.stats.IndexStatsIT > testConcurrentIndexingAndStatsRequests FAILED
01:43:58     java.lang.AssertionError: 
01:43:58     Expected: an empty collection
01:43:58          but: <[[test][7] failed, reason [FailedNodeException[Failed node [QSoO4D6ZQ56SM8kqibBAkg]]; nested: RemoteTransportException[[node_s1][127.0.0.1:37625][indices:monitor/stats[n]]]; nested: IllegalStateException[Negative longs unsupported, use writeLong or writeZLong for negative numbers [-1]]; ], [test][5] failed, reason [FailedNodeException[Failed node [QSoO4D6ZQ56SM8kqibBAkg]]; nested: RemoteTransportException[[node_s1][127.0.0.1:37625][indices:monitor/stats[n]]]; nested: IllegalStateException[Negative longs unsupported, use writeLong or writeZLong for negative numbers [-1]]; ], [test][9] failed, reason [FailedNodeException[Failed node [QSoO4D6ZQ56SM8kqibBAkg]]; nested: RemoteTransportException[[node_s1][127.0.0.1:37625][indices:monitor/stats[n]]]; nested: IllegalStateException[Negative longs unsupported, use writeLong or writeZLong for negative numbers [-1]]; ], [test][3] failed, reason [FailedNodeException[Failed node [QSoO4D6ZQ56SM8kqibBAkg]]; nested: RemoteTransportException[[node_s1][127.0.0.1:37625][indices:monitor/stats[n]]]; nested: IllegalStateException[Negative longs unsupported, use writeLong or writeZLong for negative numbers [-1]]; ], [test][1] failed, reason [FailedNodeException[Failed node [QSoO4D6ZQ56SM8kqibBAkg]]; nested: RemoteTransportException[[node_s1][127.0.0.1:37625][indices:monitor/stats[n]]]; nested: IllegalStateException[Negative longs unsupported, use writeLong or writeZLong for negative numbers [-1]]; ]]>
01:43:58         at __randomizedtesting.SeedInfo.seed([72478EA70ABABFF9:C5D6E56E887CA8C7]:0)
01:43:58         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
01:43:58         at org.junit.Assert.assertThat(Assert.java:956)
01:43:58         at org.junit.Assert.assertThat(Assert.java:923)
01:43:58         at org.elasticsearch.indices.stats.IndexStatsIT.testConcurrentIndexingAndStatsRequests(IndexStatsIT.java:1206)
01:43:58 REPRODUCE WITH: ./gradlew ':server:integTest' --tests "org.elasticsearch.indices.stats.IndexStatsIT.testConcurrentIndexingAndStatsRequests" -Dtests.seed=72478EA70ABABFF9 -Dtests.security.manager=true -Dtests.locale=sr-Latn-BA -Dtests.timezone=BET -Dcompiler.java=13
```

I am not 100% sure this is a true failure, since in the test that ran after this one there is a failing node, but at the same time the error itself in the testConcurrentIndexingAndStatsRequests  test doesn't have a root exception related to the node failure:

```
01:43:58   1> [2020-02-16T19:43:57,992][INFO ][o.e.c.c.Coordinator      ] [node_s1] master node [{node_s0}{lf2hdgwfShOQpBcXhvQ8mQ}{EkQzmawFRDeKE7Na-uro4Q}{127.0.0.1}{127.0.0.1:40999}{dim}] failed, restarting discovery
01:43:58   1> org.elasticsearch.transport.NodeDisconnectedException: [node_s0][127.0.0.1:40999][disconnected] disconnected
01:43:58   1> [2020-02-16T19:43:57,993][INFO ][o.e.c.s.ClusterApplierService] [node_s1] master node changed {previous [{node_s0}{lf2hdgwfShOQpBcXhvQ8mQ}{EkQzmawFRDeKE7Na-uro4Q}{127.0.0.1}{127.0.0.1:40999}{dim}], current []}, term: 1, version: 164, reason: becoming candidate: onLeaderFailure
01:43:58   1> [2020-02-16T19:43:57,996][WARN ][o.e.c.NodeConnectionsService] [node_s1] failed to connect to {node_s0}{lf2hdgwfShOQpBcXhvQ8mQ}{EkQzmawFRDeKE7Na-uro4Q}{127.0.0.1}{127.0.0.1:40999}{dim} (tried [1] times)
01:43:58   1> org.elasticsearch.transport.ConnectTransportException: [node_s0][127.0.0.1:40999] connect_exception
```

--- a/server/src/main/java/org/elasticsearch/ingest/IngestMetric.java
+++ b/server/src/main/java/org/elasticsearch/ingest/IngestMetric.java
@@ -22,6 +22,8 @@ package org.elasticsearch.ingest;
 import org.elasticsearch.common.metrics.CounterMetric;
 import org.elasticsearch.common.metrics.MeanMetric;
 
+import java.util.concurrent.atomic.AtomicLong;
+
 /**
  * <p>Metrics to measure ingest actions.
  * <p>This counts measure documents and timings for a given scope.
@@ -39,7 +41,7 @@ class IngestMetric {
      * The current count of things being measure. Should most likely ever be 0 or 1.
      * Useful when aggregating multiple metrics to see how many things are in flight.
      */
-    private final CounterMetric ingestCurrent = new CounterMetric();
+    private final AtomicLong ingestCurrent = new AtomicLong();
     /**
      * The ever increasing count of things being measured
      */
@@ -53,7 +55,7 @@ class IngestMetric {
      * Call this prior to the ingest action.
      */
     void preIngest() {
-        ingestCurrent.inc();
+        ingestCurrent.incrementAndGet();
     }
 
     /**
@@ -61,7 +63,7 @@ class IngestMetric {
      * @param ingestTimeInMillis The time it took to perform the action.
      */
     void postIngest(long ingestTimeInMillis) {
-        ingestCurrent.dec();
+        ingestCurrent.decrementAndGet();
         ingestTime.inc(ingestTimeInMillis);
         ingestCount.inc();
     }
@@ -90,6 +92,6 @@ class IngestMetric {
      * Creates a serializable representation for these metrics.
      */
     IngestStats.Stats createStats() {
-        return new IngestStats.Stats(ingestCount.count(), ingestTime.sum(), ingestCurrent.count(), ingestFailed.count());
+        return new IngestStats.Stats(ingestCount.count(), ingestTime.sum(), ingestCurrent.get(), ingestFailed.count());
     }
 }

